src/main/java/com/capitalone/dashboard/service/CollectorServiceImpl.java

```diff
@@ -284,6 +288,20 @@ public Collector createCollector(Collector collector) {
         return (List<CollectorItem>) collectorItemRepository.findAll(ids);
     }
 
+    @Override
+    public void deletePropertiesInCollectorById(String id) {
+        ObjectId objectId = new ObjectId(id);
+        List<Collector> collectors = collectorsById(objectId);
```

Comment:
collectorsById should return a single instance of collector and not list at is findOne and not findAll.

Reply:
collectorsById -> findById -> returns list (existing CollectorRepository class method)

Reply:
@nireeshT changed in core and api

Reply:
@nireeshT used findOne instead, but in the future, we may want to change findById to return overall one Collector vs. list.